project/contracts/RandomRarity.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import "hardhat/console.sol";
+
+/**
+    @dev Simple smart contract that allows you to create NFT collections.
+    You create the collection and add items (NFTs) to the collection.
+    Then you can query all metadata of a certain collection via 1 call.
+    Notes:
+    -   1 NFTAlbum (Owned by a user) -> 1..N Collections
+    -   1 Collection -> 1..N Items
+ */
+contract RandomRarity is VRFConsumerBase {
+
+    bytes32 internal keyHash;
+    uint256 internal fee;           // link token fee for oracle
+    uint256 internal randomResult;
+
+    enum Rarity{ 
+        COMMON, 
+        RARE, 
+        EPIC, 
+        LEGENDARY 
+    }
+
+    // numbers are out of 10,000 so we can compare ints
+    uint constant commonChance = 7223;
+    uint constant rareChance = 2200;
+    uint constant epicChance = 457;
+    uint constant legendaryChance = 12;
+    uint256 constant uint256Max = 2**35 - 1;
```

Comment:
This is poorly named here, but represents the max for the initial random number generator.